pkg/api/platformapi/deploymentapi/params.go

```diff
@@ -0,0 +1,195 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package deploymentapi
```

Comment:
This will collide with `pkg/api/deploymentapi`, maybe it makes sense to rename it to `configurationtemplateapi`? The name isn't great, but I think the package itself doesn't quite match the API space, `PlatformConfigurationTemplates` is the object used to store that API path within the auto-generated models, what do you think?

Reply:
argh! of course! let me change that